### PR TITLE
Added support for setting the legend font size

### PR DESCRIPTION
**Pull request by jlstevens**
This PR adds support for setting the fontsize in legends, addressing issue #353.

It uses the same system as everything else, i.e:

```python
%%opts NdOverlay [fontsize={'legend':14}]
hv.NdOverlay({i:hv.Curve(np.random.rand(10,2)) for i in range(3)})
```
image

### CODE REVIEW

**Comment by jbednar** [comment]
Looks very straightforward; I'd be happy to merge once the tests pass.  It would be nice to mention how to change fontsizes in the tutorials; if only that were a quick edit followed by clicking 'ok' to update the reference data!

**Comment by jlstevens** [comment]
The commit above also make it possible to set the font size of the legend title using the key ``legend_title``. Note this currently applies to the matplotlib backend only as the bokeh backend doesn't have legend titles (Philipp is looking into this).

**Comment by jlstevens** [comment]
> It would be nice to mention how to change fontsizes in the tutorials

Agreed. Which tutorial would you mention it in? Maybe the containers tutorial?

I need to update the test data anyway because of the options tutorial (see issue #89, it gets really annoying!).

**Comment by sf-issues** [comment]
Probably the options tutorial, since this is clearly an option that
people will want to change.

-- 
The University of Edinburgh is a charitable body, registered in
Scotland, with registration number SC005336.

**Comment by jlstevens** [comment]
``sf-issues`` is alive! It has gained sentience! ;-p

But yes, the options tutorial makes the most sense.

**Comment by jlstevens** [comment]
It was tricky to find a nice play to insert an example of using ``fontsize`` in the Options tutorial. In the commit above, I think I found a way to insert it naturally into the tutorial while expanding on the content in a useful  way.

@jbednar @philippjfr I would appreciate it if you could have a look and give me feedback before I update the testing data.

File: doc/Tutorials/Options.ipynb [hunk 1]
```diff
@@ -523,6 +524,45 @@
...
+   "source": [
+    "%%opts Curve (color='r') [fontsize={'xlabel':15, 'ticks':8}] \n",
+    "layout"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The color of the curve has been changed to red and the fontsizes of the x-axis label and all the tick labels has been modified. The ``fontsize`` is an important plot option and you can find more information about the available options in the ``fontsize`` documentation above.\n",
```
Comment (jbednar) [inline doc/Tutorials/Options.ipynb]: s/has/have

**Comment by jbednar** [comment]
That explains it!  In July I switched my github account to my new email address, but the effect was that I now got two copies of every single issue, one to the old and one to the new address.  I couldn't find any mention of my old address in my account, so I was completely mystified, but the reason must be that my old address is also listed in the sf-issues account!  Now I can finally get rid of that pesky deja-vu feeling that starts every day. :-)

File: doc/Tutorials/Options.ipynb (hunk 1 above)
Comment (jlstevens) [inline doc/Tutorials/Options.ipynb]: There is one curve shown so 'have' doesn't seem right to me...

**Comment by jlstevens** [comment]
Glad I could help! :-)

I'm not quite sure about the comment, but as soon as that is resolved, I'll update the reference data and merge.

File: doc/Tutorials/Options.ipynb (hunk 1 above)
Comment (jbednar) [inline doc/Tutorials/Options.ipynb]: "fontsizes ... has" -> "fontsizes ... have"

Comment (jlstevens) [inline doc/Tutorials/Options.ipynb]: I see it now, thanks! I'll fix that in a minute.

**Comment by philippjfr** [comment]
Looks good, merging now.

**Pull request merged**